READMEにDB設計追記＆ER図作成

t-hosakaa opened the pull request:
# What
・各モデル同士のアソシエーション記述
・外部キーは親子関係にあるモデルのうち、子に当たるモデルのみに記述。（親に当たる方のテーブルからは削除しています）

# Why
フリマアプリを開発する上で必要なテーブル（モデル）と、それぞれの関係性（アソシエーション）を明確にしておくため。
開発初期のこの段階でDB設計を明確にしておくことで今後の開発をスムーズに進めることができるから。

t-hosakaa committed:
Update README.md②

・各モデル同士のアソシエーション記述
・外部キーは親子関係にあるモデルのうち、子に当たるモデルのみに記述。（親に当たる方のテーブルからは削除しています）

t-hosakaa committed:
Update README.md

t-hosakaa committed:
Update README.md

t-hosakaa committed:
Update README.md

・Itemsテーブルの「comment_id」は不要なので削除。
・Itemsテーブルに「size」カラムを追加。
・新たにSub_Categoriesテーブル、Likesテーブルを作成

t-hosakaa committed:
Update README.md

・「Items_Categories」→「Item_Categories」テーブルに名称変更

t-hosakaa commented:
コードレビューをお願いします。
①READMEにDB設計を追記
②DB設計のER図作成
URL：https://drive.google.com/file/d/1hACnkDk7tSwZfzW4wEQgeccvvLQNGR6P/view?usp=sharing

ghost commented:
お疲れ様です。大塚です。
LGTM

naru515 commented:
ありとうございます！
LGTM!

amy5-5-5 commented:
ありがとうございます！
LGTMです！！

hiroya418 commented:
遅くなり申し訳ございません、LGTMです

miyagit commented on README.md:
```diff
@@ -39,65 +38,118 @@ Things you may want to cover:
+* has_many comments
+* has_many likes
 
+## Credit_cardsテーブル
+|Column|Type|Options|
+|------|----|-------|
+|card_number|integer|null: false|
+|exp_date|date|null: false|
+|security_number|integer|null: false|
+|name|string|null: false|
+|user_id|references|foreign_key: true|
```
こちらのテーブルは今回必要ないので削除しましょう。それに付随するアソシエーションの削除もお願いします。

t-hosakaa replied:
コードレビューありがとうございます。
質問です。Credit_cardsテーブルについては商品購入の際に必要ではないでしょうか？（Trelloに記載の仕様参照しました）現時点では必要無いということでしょうか？

miyagit replied:
理由としまして、今回の実装では、itemを購入するたびにpayjp側にcard情報を入力するためです。またcard情報をDBに保存するのはセキュリティ的にも危ないため、payjp側にcard情報に関する処理をまかせたいためです。

下記のカリキュラムにpayjpの実装について詳しく載っているので、良ければ参考にして頂ければと思います。
https://master.tech-camp.in/curriculums/4754

t-hosakaa replied:
ご回答ありがとうございます。承知しました。
Credit_cardsテーブルの削除とそれに付随するアソシエーションの削除をしましたのでご確認お願いいたします。

miyagit reviewed (commented):
Credit_cardsテーブル、Commentsテーブル、Images、Likesテーブルは

t-hosakaa committed:
Update README

Credit_cardsテーブルの削除とそれに付随するアソシエーションの削除

miyagit reviewed (commented):
Category、SubCategory、ItemCategoryとありますが、Categoryテーブルだけにしましょう。

Categoryの階層構造に関しては、ancestryを使って実装しましょう。
ancestryのgemに関しては、下記のサイトに詳しく載っているので、よければ参考にしてください。

https://qiita.com/Sotq_17/items/120256209993fb05ebac

t-hosakaa committed:
Update README.md

・Item_categoriesテーブル（中間テーブル）を削除
・CategoriesテーブルとItemsテーブルの関係が1対多となるようにアソシエーションの記述変更
・Gemfileに「gem 'ancestry'」追記、「bundle install」
・Categoriesのアソシエーションに「has_ancestry」追記しancestryを使用することを明記

t-hosakaa commented:
ご指摘ありがとうございます。
修正しましたので再度ご確認お願いいたします。

miyagit reviewed (approved):
LGTMです！変更点が出れば随時修正していきましょう！

t-hosakaa committed:
gem 'ancestry'

t-hosakaa committed:
Merge branch 't-hosakaa-patch-1（Update-README.md）' of https://github.com/naru515/fleamarket_sample_78a into t-hosakaa-patch-1（Update-README.md）
